### PR TITLE
Add ability to get the aggsig additional data from full node RPC

### PR DESCRIPTION
**Pull request by matt-o-how**
### Purpose:
This adds in the ability to request a full node to tell you what the additional data required for an aggsig_me is.
This makes it easier for people to create applications which support use in the full node and create signatures to work on testnet, simulator, mainnet, and even other chia forks.

### Current Behavior:
Function does not currently exist.

### New Behavior:
Added `get_aggsig_additional_data` to both RPC API and RPC Client.

### Testing Notes:
Added a line of test in to RPC tests.

### CODE REVIEW

**Comment by matt-o-how** [comment]
Despite the branch name, this code no longer adds the ability to melt CATs

File: chia/rpc/full_node_rpc_client.py [hunk 1]
```diff
@@ -159,6 +159,12 @@ async def get_coin_records_by_parent_ids(
         response = await self.fetch("get_coin_records_by_parent_ids", d)
         return [CoinRecord.from_json_dict(coin_record_dict_backwards_compat(coin)) for coin in response["coin_records"]]
 
+    async def get_aggsig_additional_data(self) -> str:
```
Comment (Quexington) [inline chia/rpc/full_node_rpc_client.py:162]: Let's return bytes from this client endpoint.  `bytes32` if it's appropriate.

File: chia/rpc/full_node_rpc_client.py [hunk 2]
```diff
@@ -159,6 +159,12 @@ async def get_coin_records_by_parent_ids(
...
+        result = await self.fetch("get_aggsig_additional_data", {})
+        add_data: str = result["additional_data"]
+        assert add_data is not None
```
Comment (Quexington) [inline chia/rpc/full_node_rpc_client.py:165]: This is not necessary according to the hint above so one of them is wrong.  I don't think you need to assign this to a variable though, just return `hexstr_to_bytes` (or `bytes32.from_hexstr`) on `result["additional_data"]`.

**Review by Quexington — CHANGES_REQUESTED** [review]

File: chia/rpc/full_node_rpc_client.py (hunk 2 above)
Comment (matt-o-how) [reply chia/rpc/full_node_rpc_client.py]: Done

File: chia/rpc/full_node_rpc_client.py (hunk 1 above)
Comment (matt-o-how) [reply chia/rpc/full_node_rpc_client.py]: Done

**Review by Quexington — APPROVED** [review]

**Review by arvidn — APPROVED** [review]

**Pull request merged**